attendance and client visit report pages

lathamaniTech opened the pull request:
created apiService file and created component for subheader

lathamaniB committed:
attendance and client visit report

attendance and client visit

lathamaniB committed:
Merge branch 'main' of https://github.com/KarthickTechie/LMSRN

lathamaniB committed:
Update attendance.tsx

lathamaniB committed:
Merge branch 'KarthickTechie:main' into main

lathamaniB committed:
author, date added

author, date, description added

KarthickTechie commented on lib/appwrite.js:
please organize gps , geocoding functions in isolated service , don't add any code appwrite.js. This  file is used for appwrite platform only

lathamaniTech replied:
resolved. moved to new file

KarthickTechie commented on lib/appAPIServices.ts:
```diff
@@ -0,0 +1,35 @@
+      reqBody,
+      { headers: options }
+    );
+    console.log(data, "response");
```
use console.info

KarthickTechie commented on .expo/types/router.d.ts:
router.d.ts - .gitignore this. libraray files should not be commited

lathamaniTech replied:
o., i will not commit those files next time

KarthickTechie commented on app.json:
```diff
@@ -14,25 +14,38 @@
+      "package": "com.sysarc.lms",
+      "config": {
+        "googleMaps": {
+          "apiKey": "AIzaSyD6dBKMTE52g9AwNHYxOQ9GejJ8x8-YrhI"
```
api key should not be added in git public repo . we can configure it system environment variable

KarthickTechie commented on app/(home)/attendance.tsx:
```diff
@@ -1,14 +1,191 @@
+import MapView, { MapMarker } from "react-native-maps";
+import { router } from "expo-router";
+import * as AppType from "@/apptypes/AppTypes";
+import { getLocationCoordinates } from "@/lib/appwrite";
```
getLocaltionCoordinates should be moves to different services file .

lathamaniTech replied:
created separate file

KarthickTechie commented on app/(home)/visitreport.tsx:
```diff
@@ -1,12 +1,330 @@
+    try {
+      setIsLoading(true);
+      const userId = "55655";
+      const reqBody: AppType.ClientVisitRequest = {
```
use pure functions for object constructions

lathamaniTech replied:
done

KarthickTechie commented on app/(home)/visitreport.tsx:
```diff
@@ -1,12 +1,330 @@
+        `${AppType.APIMethods.saveClientVisit}/${userId}`,
+        reqBody
+      );
+      console.log("fjkdjs", resp);
```
use console.info , never use console.log

lathamaniTech replied:
sure. next time i will not repeat

KarthickTechie commented on app/(home)/visitreport.tsx:
```diff
@@ -1,12 +1,330 @@
+        reqBody
+      );
+      console.log("fjkdjs", resp);
+      if (resp.status == "200 OK") {
```
use enum for response status

lathamaniTech replied:
done

KarthickTechie reviewed (commented):
pls resolve the review comments

lathamaniB committed:
Pull Request Review Comments and LOV data map

created separate file for geoloation function, map static data to lov fields. in visit report

lathamaniB committed:
removed api key from app.json

removed api key and created env variables for map api key and api url and created app.config.ts file for google map key access
